### PR TITLE
🗿 Fix importing/copying flows when flow has group dependency with no name

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/contacts/models.py [hunk 1]
```diff
@@ -2696,7 +2696,7 @@ class ContactGroup(TembaModel):
     user_groups = UserContactGroupManager()
 
     @classmethod
-    def get_user_group(cls, org, name):
+    def get_user_group_by_name(cls, org, name):
```
Comment (rowanseymour) [inline temba/contacts/models.py]: just renaming for clarity while I'm in here

**Comment by nicpottier** [comment]
So is this going to get us in a state of having definitions we can't import because those defs don't have names?

**Comment by rowanseymour** [comment]
@nicpottier no.. this is supposed to ensure that we match groups by UUID regardless of whether there is a name

**Comment by nicpottier** [comment]
But just to be clear, if we import a definition which currently is written without a name, wouldn't this fail in `get_or_create`?

**Comment by rowanseymour** [comment]
Yeah, if someone tried to import a new flow definition, with a `has_group`, created by the editor since the migrated ones have names, and the group doesn't exist, we'd blow up at the assert in `get_or_create` and the import would fail.

Only alternative would be to generate a new unique name for the group... which is maybe not terrible?

**Comment by nicpottier** [comment]
Right, ya, I mean this is the case we are seeing ya? So this would make that import fail and just worried if we have others of those in the wild. Not super sure what the right thing to do here is. Name it the same as the UUID? I feel like that's *probably* better than disallowing the import entirely.

**Comment by rowanseymour** [comment]
Not quite - in this case the group exists. The user is trying to copy a valid flow whose dependencies exist, it's just the `has_group` doesn't have a name so because of the `and "name" in ref` we don't bother recording it in the dependency mapping.

To be clear most users aren't going to run into this because exports include groups now.. and old exports have legacy exports have legacy flow definitions which don't have this problem because the group name is always there.

We should tho fix https://github.com/nyaruka/floweditor/issues/687

**Comment by nicpottier** [comment]
Alright, fair enough. We can reconsider if we see this occur more somehow.

**Pull request merged**